### PR TITLE
(Hopefully) Fixes crashing bug

### PR DESCRIPTION
**Pull request by ColeoCofer**
This should have hopefully fixed the bug.
Moved the audio recording prompt into MainActivity onCreate().
Waits to enable the search button until spotify is logged in.

### CODE REVIEW

File: app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java [hunk 1]
```diff
@@ -151,14 +134,23 @@ protected void onPause() {
...
     public void onWaveFormDataCapture(Visualizer visualizer, byte[] waveform, int samplingRate) {
     }
 
     @Override
     public void onFftDataCapture(Visualizer visualizer, byte[] fft, int samplingRate) {
         /** Gives us the decibel level for the fft bucket we care about **/
-        double dbs = utils.getDBs(fft[REAL_BUCKET], fft[IMAGINARY_BUCKET], this.audioSampleSize);
+        double dbs = getDBs(fft[REAL_BUCKET], fft[IMAGINARY_BUCKET], this.audioSampleSize);
+
+        //this.previousUpdateTime = updateDbHistory(dbs, VisualizerModel.getInstance().currentVisualizer.dbHistory, this.previousUpdateTime);
```
Comment (DanielEliasBermudez) [inline app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java]: this should be deleted if its not being used

File: app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java [hunk 2]
```diff
@@ -151,14 +134,23 @@ protected void onPause() {
         surfaceView.onPause();
     }
 
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        Log.d("test", "IM BEING DESTROYED");
```
Comment (DanielEliasBermudez) [inline app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java]: this should be deleted

File: app/src/main/java/com/example/colecofer/android_audio_visualizer/Utility.java [hunk 3]
```diff
@@ -15,4 +22,40 @@ static double getDBs(byte real, byte imaginary, int n) {
...
+     * removes the oldest record. If the current dB level exceeds our max setting it uses the max
+     * @param newDB
+     * @param dbHistory
+     */
+    static long updateDbHistory(double newDB, ArrayDeque<Float> dbHistory, long previousUpdateTime) {
+        Pair<Long, Boolean> isTime = isTimeToUpdate(previousUpdateTime);
+
+        if (isTime.second == true) {
+            float dbRatio = (float) newDB / MAX_DB_LEVEL;
+            dbRatio = dbRatio > 1.0f ? 1.0f : dbRatio;
```
Comment (DanielEliasBermudez) [inline app/src/main/java/com/example/colecofer/android_audio_visualizer/Utility.java]: we should make 1.0f a variable with a name. something like maxDecibalValue

File: app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java (hunk 2 above)
Comment (ColeoCofer) [reply app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java]: Fixed.

File: app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java (hunk 1 above)
Comment (ColeoCofer) [reply app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java]: Leaving for the future

**Pull request merged**